Newsletter for January 2025 (#384)
maplibre/maplibre.github.io: src/content/news/2025-02-03-maplibre-newsletter-january-2025/index.mdx

---
title: "MapLibre Newsletter January 2025"
date: "2025-02-03"
categories: ["newsletter"]
authors:
  [bart, birk, ianthetechie, yuri, harel, brandon, josxha, ramyaragupathy]
draft: false
---

import { Image } from "astro:assets";
import gratitude_aws from "./aws-2024-sponsor.png";

As we step into 2025, we want to start with a heartfelt thank you to AWS for renewing their sponsorship for MapLibre. Their ongoing commitment since 2022 has played a vital role in the growth, and strengthening of our global community.

<div style={{ display: "flex", justifyContent: "center" }}>
  <Image
    src={gratitude_aws}
    style={{
      maxWidth: "550px",
      width: "100%",
      height: "auto",
      display: "inline-block",
    }}
    alt="Thanks to AWS sponsorship renewal!"
  />
</div>
<br></br>

> In addition to AWS, we want to extend our gratitude to all our supporters—both individuals and organizations—who contribute through GitHub Sponsors and OpenCollective. We acknowledge the collective contributions of our community, and we are grateful for your continued backing.

Reviewer: perhaps link to the opencollective donation page here - for transparency?

```suggestion
In addition to AWS, we want to extend our gratitude to all our supporters—both individuals and organizations—who contribute through GitHub Sponsors and [OpenCollective](https://opencollective.com/maplibre). We acknowledge the collective contributions of our community, and we are grateful for your continued backing.
```

With that note, here's the round-up of updates from January 2025!

## MapLibre Native

> - [MapLibre Android 11.8.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.0) and [MapLibre iOS 6.10.0 ](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.10.0) were released which included the new PMTiles support for MapLibre Native.
> - The MapLibre Native Development Team kicked off their work again this January. They are currently working on:
>   - Extending Custom Drawable Layer support for cross-platform custom layers.
>   - Using a dynamic texture atlas for reduced memory usage.
>   - Adding support for [MapLibre Tiles](https://github.com/maplibre/maplibre-tile-spec) which we expect will bring better performance.
>   - Ongoing performance and stability improvements.
> - We want to bring Vulkan and Metal support to Qt. Qt is currently using an older `opengl2` branch. If you are a [MapLibre Native Qt](https://github.com/maplibre/maplibre-native-qt) user and want to help, please see [this draft PR](https://github.com/maplibre/maplibre-native-qt/pull/59).
> - [@TatyanaPolunina](https://github.com/TatyanaPolunina), a C++ engineer from Lyft, will be addressing some stability issues Lyft is facing, well as extending the observability of MapLibre Native. We are happy to work with Tetiana and Lyft on MapLibre Native this year!
> - MapLibre Native gained experimental Rust support ([docs](https://maplibre.org/maplibre-native/docs/book/rust.html), [PR](https://github.com/maplibre/maplibre-native/pull/3137)). This allows C++ code to use Rust modules including dependencies. The first demo replaces csscolorparser.hpp with a popular [Rust lib](https://crates.io/crates/csscolorparser) (2.2 million downloads). We need feedback on any potential issues with this approach. Thanks to [@louwers](https://github.com/louwers) and [@nyurik](https://github.com/nyurik).
> - Work [has begun](https://github.com/maplibre/maplibre-native/pull/3181) on MapLibre Native Rust wrapper - the ability to use native lib as a Rust crate. This will add another supported platform, in addition to Linux/Android/iOS/Node.js and others.

Reviewer: lets put what we have already done before what we plan to do?  The first point is what we did, the next two are about plans. Lets reorder the above 5 points as: 1,4,2,3,5

```suggestion
- [MapLibre Android 11.8.0](https://github.com/maplibre/maplibre-native/releases/tag/android-v11.8.0) and [MapLibre iOS 6.10.0 ](https://github.com/maplibre/maplibre-native/releases/tag/ios-v6.10.0) were released which included the new PMTiles support for MapLibre Native.
- MapLibre Native gained experimental Rust support ([docs](https://maplibre.org/maplibre-native/docs/book/rust.html), [PR](https://github.com/maplibre/maplibre-native/pull/3137)). This allows C++ code to use Rust modules including dependencies. The first demo replaces csscolorparser.hpp with a popular [Rust lib](https://crates.io/crates/csscolorparser) (2.2 million downloads). We need feedback on any potential issues with this approach. Thanks to [@louwers](https://github.com/louwers) and [@nyurik](https://github.com/nyurik).
- The MapLibre Native Development Team kicked off their work again this January. They are currently working on:
  - Extending Custom Drawable Layer support for cross-platform custom layers.
  - Using a dynamic texture atlas for reduced memory usage.
  - Adding support for [MapLibre Tiles](https://github.com/maplibre/maplibre-tile-spec) which we expect will bring better performance.
  - Ongoing performance and stability improvements.
- We want to bring Vulkan and Metal support to Qt. Qt is currently using an older `opengl2` branch. If you are a [MapLibre Native Qt](https://github.com/maplibre/maplibre-native-qt) user and want to help, please see [this draft PR](https://github.com/maplibre/maplibre-native-qt/pull/59).
- [@TatyanaPolunina](https://github.com/TatyanaPolunina), a C++ engineer from Lyft, will be addressing some stability issues Lyft is facing, well as extending the observability of MapLibre Native. We are happy to work with Tetiana and Lyft on MapLibre Native this year!
- Work [has begun](https://github.com/maplibre/maplibre-native/pull/3181) on MapLibre Native Rust wrapper - the ability to use native lib as a Rust crate. This will add another supported platform, in addition to Linux/Android/iOS/Node.js and others.
```

Highlights of this release include:

- **New Architecture:** The SDK has not been refactored to use New Architecture yet, so there isn't any performance improvement, but v10 includes a new compat layer that allows MapLibre Native to be used in New Architecture mode.

- **Expo Support:** v10 includes improvements for use in Expo, such as the ability to add Google Location Services.

- **Flexible SDK Versioning:** This release allows users to configure which version of the iOS and Android SDKs from the MapLibre Native distributions are used. This works for minor and patch releases, enabling users of MapLibre React Native to adopt these new versions immediately after they are published, without waiting for a MapLibre React Native release.

- **New Documentation:** https://maplibre.org/maplibre-react-native/

This release has been a year in the making, with 29 alphas and 21 betas! Thanks to [@tyrauber](https://github.com/tyrauber) and [@KiwiKilian](https://github.com/KiwiKilian) for their contributions, helping to enhance its capabilities and improve documentation.

## Web

> The year started with a bang 🎉. With the [release of the globe](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.0.0) almost on New Year's eve we were off to a great start in 2025. A very important discussion started with the release of version 5 and the change we decided to do with `geometry-type`, which led to the trust of 5.0.1 soon after and the reversion of a breaking change in the style.

Reviewer: `lead to the trust` - is that a british expression? :)

```suggestion
The year started with a bang 🎉. With the [release of the globe](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.0.0) almost on New Year's eve we were off to a great start in 2025. One breaking change related to `geometry-type` in the style spec caused a lot of important discussion, resulting in a quick revert of that specific change as 5.0.1.
```

#### MapLibre Web Fonts

A group of developers using MapLibre GL JS is interested in building native Web Fonts (`.woff2`) support, as a replacement for needing to fetch and host `glyphs`. This would have key advantages:

- Fewer font files and less data to download when displaying labels.
- Easier workflow for designing maps with custom typography.
- Support for more languages in South and Southeast Asia.

A demonstration is live [here](https://bdon.github.io/maplibre-text/webfonts/).

You can find more information on [GitHub Discussions](https://github.com/maplibre/maplibre-gl-js/discussions/2555) or in the `#maplibre-i18n` slack channel.

## MapLibre Flutter

- The MapLibre Native Android SDK has been updated to 11.8.0 which adds support for PMTiles. It is planned to add support for PMTiles on web as well to provide feature parity across all platforms.
- Static-typed map styles, including expressions have been evaluated. Modeling the MapLibre style spec with JSON schema to make the code generation more straight forward looks promising.

> 

Reviewer: ```suggestion

## Martin Tile Server

Despite announcing v0.15 Martin release last month, we hit a last second testing issue that delayed the release until mid January.  Moreover, additional testing uncovered some suprising inconsistencies with the newer PostgreSQL versions itself (unrelated to Martin).  That said, Martin v0.15 is out, thanks to the amazing volunteer work of the team.

```

## Monthly meetings

We will hold our monthly meeting on the second Wednesday of each month and run an additional session on the last Wednesday to better accommodate participants in Asia time zones.

#### Second Wednesday Meetings (Feb 12, 2025)

- MapLibre Navigation: 5:00–6:00 PM UTC
- MapLibre Native: 6:00–7:00 PM UTC
- MapLibre GL JS: 7:00–8:00 PM UTC

> #### Last Wednesday Meeting ( Asia Friendly)

Reviewer: ```suggestion
#### Last Wednesday Meeting (Asia Friendly)
```

---
title: "MapLibre Newsletter January 2025"
date: "2025-02-03"
categories: ["newsletter"]
authors: [bart, birk, ianthetechie, yuri, harel, brandon, josxha, ramyaragupathy]
draft: false
---

import { Image } from "astro:assets";
import gratitude_aws from "./aws-2024-sponsor.png";

> As we step into 2025, we want to start with a heartfelt thank you to AWS for renewing their sponsorship for MapLibre. Their ongoing commitment since 2022 has played a vital role in the growth, and strengthening of our global community.

Reviewer: We should change this a bit to also mention smaller contributors, maybe just by name or for the smallest ones maybe aggregate them somehow?  I do not want us to appear as only receiving sponsorships from the large organizations... I do want to thank AWS for the large portion of the donations though, so they should be front and center.

Author: For this edition, I will go with a line acknowledging the contributions in general. In the coming months I will look at improving this content further.

Highlights of this release include:

- **New Architecture:** The SDK has not been refactored to use New Architecture yet, so there isn't any performance improvement, but v10 includes a new compat layer that allows MapLibre Native to be used in New Architecture mode.

- **Expo Support:** v10 includes improvements for use in Expo, such as the ability to add Google Location Services.

- **Flexible SDK Versioning:** This release allows users to configure which version of the iOS and Android SDKs from the MapLibre Native distributions are used. This works for minor and patch releases, enabling users of MapLibre React Native to adopt these new versions immediately after they are published, without waiting for a MapLibre React Native release.

- **New Documentation:** https://maplibre.org/maplibre-react-native/

This release has been a year in the making, with 29 alphas and 21 betas! Thanks to [@tyrauber](https://github.com/tyrauber) and [@KiwiKilian](https://github.com/KiwiKilian) for their contributions, helping to enhance its capabilities and improve documentation.

## Web

> The year started with a bang 🎉. With the [release of the globe](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.0.0) almost on New Year's eve we were off to a great start in 2025. One breaking change related to `geometry-type` in the style spec caused a lot of important discussion, resulting in a quick revert of that specific change as 5.0.1.

Reviewer 2: ```suggestion
The year started with a bang 🎉. With the [release of the globe](https://github.com/maplibre/maplibre-gl-js/releases/tag/v5.0.0) almost on New Year's eve we were off to a great start of 2025. One breaking change related to `geometry-type` in the style spec caused a lot of commotion and inspired an important discussion, resulting in a quick revert of that specific change with the release of 5.0.1.
```

> While this discussion was valuable we encourage the community to use our pre-releases for major version and provide feedback before the official release.

Reviewer 2: ```suggestion
While this discussion was valuable we encourage the community to use our pre-releases for major versions and provide feedback before the official release.
```
